ReactAndroid/src/main/java/com/facebook/react/NativeModuleRegistryBuilder.java

```diff
@@ -114,19 +103,28 @@ public void processPackage(ReactPackage reactPackage) {
   public void addNativeModule(NativeModule nativeModule) {
     String name = nativeModule.getName();
     Class<? extends NativeModule> type = nativeModule.getClass();
-    if (namesToType.containsKey(name)) {
-      Class<? extends NativeModule> existingModule = namesToType.get(name);
-      if (!nativeModule.canOverrideExistingModule()) {
+    boolean canOverride = nativeModule.canOverrideExistingModule();
+
+    putModuleTypeAndHolderToModuleMaps(type, name, canOverride, new ModuleHolder(nativeModule));
+  }
+
+  private void putModuleTypeAndHolderToModuleMaps(Class<? extends NativeModule> type, String underName,
+                                                  boolean canOverrideExistingModule,
```

Comment:
Looks like `ModuleHolder` has a `getCanOverrideExistingModule` method, we can use this inside of putModuleTypeAndHolderToModuleMaps instead of passing canOverrideExistingModule as a parameter.